config/commands.js

```diff
@@ -1100,24 +1100,67 @@ var commands = exports.commands = {
 		var all = (cmd === 'learnall');
 		if (cmd === 'learn5') lsetData.set.level = 5;
 		if (cmd === 'g6learn') lsetData.format = {noPokebank: true};
+		var buffer = template.name;
 
-		if (!template.exists) {
-			return this.sendReply("Pokemon '" + template.id + "' not found.");
-		}
-
-		if (targets.length < 2) {
-			return this.sendReply("You must specify at least one move.");
-		}
+		if (!template.exists) return this.sendReply("Pokemon '" + template.id + "' not found.");
+		if (targets.length < 2) return this.sendReply("You must specify at least one move.");
 
 		for (var i = 1, len = targets.length; i < len; ++i) {
 			move = Tools.getMove(targets[i]);
 			if (!move.exists) {
-				return this.sendReply("Move '" + move.id + "' not found.");
+				var matchedMoves = [];
+				var targetType = '';
+				var lset = template.learnset;
+				target = targets[i].trim().toLowerCase();
+
+				var typeIndex = target.indexOf(" type");
+				if (typeIndex > -1) {
+					targetType = target.substr(0, typeIndex).capitalize();
+					if (!(targetType in Tools.data.TypeChart)) return this.sendReply("Type '" + Tools.escapeHTML(targetType) + "' not found.");
```

Comment:
Use <code>Tool.getType</code> and check <code>.exists</code> rather than sugarjs' <code>capitalize</code> and <code>in</code>.